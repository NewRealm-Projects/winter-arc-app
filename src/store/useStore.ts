import { create } from 'zustand';
import { User, DailyTracking } from '../types';

interface AppState {
  user: User | null;
  setUser: (user: User | null) => void;

  tracking: Record<string, DailyTracking>;
  setTracking: (tracking: Record<string, DailyTracking>) => void;
  updateDayTracking: (date: string, data: Partial<DailyTracking>) => void;

  darkMode: boolean;
  toggleDarkMode: () => void;

  isOnboarded: boolean;
  setIsOnboarded: (value: boolean) => void;
}

// Load dark mode preference from localStorage
const getInitialDarkMode = (): boolean => {
  try {
    const stored = localStorage.getItem('darkMode');
    if (stored !== null) {
      return JSON.parse(stored);
    }
    // Default to system preference if no stored value
    return window.matchMedia('(prefers-color-scheme: dark)').matches;
  } catch {
    return false;
  }
};

export const useStore = create<AppState>((set) => ({
  user: null,
  setUser: (user) => set({ user }),

  tracking: {},
  setTracking: (tracking) => set({ tracking }),
  updateDayTracking: (date, data) =>
    set((state) => ({
      tracking: {
        ...state.tracking,
        [date]: {
          ...state.tracking[date],
          ...data,
        },
      },
    })),

  darkMode: getInitialDarkMode(),
  toggleDarkMode: () =>
    set((state) => {
      const newDarkMode = !state.darkMode;
      try {
        localStorage.setItem('darkMode', JSON.stringify(newDarkMode));
      } catch (error) {
        console.error('Failed to save dark mode preference:', error);
      }
      return { darkMode: newDarkMode };
    }),

  isOnboarded: false,
  setIsOnboarded: (value) => set({ isOnboarded: value }),
}));
