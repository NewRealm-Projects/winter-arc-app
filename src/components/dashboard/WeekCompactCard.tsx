import {
  format,
  startOfWeek,
  addDays,
  isSameDay,
  addWeeks,
  differenceInCalendarWeeks,
  parseISO,
  isValid,
  differenceInCalendarDays,
  isAfter,
} from 'date-fns';
import { de, enUS } from 'date-fns/locale';
import { useEffect, useState } from 'react';
import { useStore } from '../../store/useStore';
import { useTranslation } from '../../hooks/useTranslation';
import { countActiveSports } from '../../utils/sports';
import { useCombinedTracking } from '../../hooks/useCombinedTracking';

export default function WeekCompactCard() {
  const { t, language } = useTranslation();
  const combinedTracking = useCombinedTracking();
  const user = useStore((state) => state.user);
  const selectedDate = useStore((state) => state.selectedDate);
  const setSelectedDate = useStore((state) => state.setSelectedDate);

  const today = new Date();
  const todayKey = format(today, 'yyyy-MM-dd');
  const activeDate = selectedDate || todayKey;
  const locale = language === 'de' ? de : enUS;

  const parseActiveDate = () => {
    if (!activeDate) {
      return today;
    }
    const parsed = parseISO(activeDate);
    return isValid(parsed) ? parsed : today;
  };

  const [weekOffset, setWeekOffset] = useState(() => {
    const todayStart = parseISO(todayKey);
    const parsedDate = activeDate ? parseISO(activeDate) : null;
    const validDate = parsedDate && isValid(parsedDate) ? parsedDate : todayStart;
    return differenceInCalendarWeeks(
      startOfWeek(validDate, { weekStartsOn: 1 }),
      startOfWeek(todayStart, { weekStartsOn: 1 }),
      { weekStartsOn: 1 }
    );
  });

  useEffect(() => {
    const todayStart = parseISO(todayKey);
    const parsedDate = activeDate ? parseISO(activeDate) : null;
    const validDate = parsedDate && isValid(parsedDate) ? parsedDate : todayStart;
    const offset = differenceInCalendarWeeks(
      startOfWeek(validDate, { weekStartsOn: 1 }),
      startOfWeek(todayStart, { weekStartsOn: 1 }),
      { weekStartsOn: 1 }
    );
    setWeekOffset(offset);
  }, [activeDate, todayKey]);

  const currentWeekStart = startOfWeek(today, { weekStartsOn: 1 });
  const displayedWeekStart = addWeeks(currentWeekStart, weekOffset);
  const displayedWeekEnd = addDays(displayedWeekStart, 6);

  const activeDateObj = parseActiveDate();
  const activeWeekStart = startOfWeek(activeDateObj, { weekStartsOn: 1 });
  const activeDayIndex = Math.min(
    Math.max(differenceInCalendarDays(activeDateObj, activeWeekStart), 0),
    6
  );

  const weekRangeFormat = language === 'de' ? 'd. MMM' : 'MMM d';
  const weekRangeLabel = `${format(displayedWeekStart, weekRangeFormat, { locale })} – ${format(
    displayedWeekEnd,
    weekRangeFormat,
    { locale }
  )}`;

  const isoWeekNumber = format(displayedWeekStart, 'I');
  const headingLabel = (() => {
    if (weekOffset === 0) {
      return t('dashboard.weekOverview');
    }
    if (weekOffset === -1) {
      return t('dashboard.lastWeek');
    }
    return t('dashboard.weekNumberTitle', { week: isoWeekNumber });
  })();

  const disableNextWeek = weekOffset >= 0;

  const handleWeekChange = (direction: 'previous' | 'next') => {
    if (direction === 'next' && disableNextWeek) {
      return;
    }

    const delta = direction === 'previous' ? -1 : 1;
    const newWeekOffset = weekOffset + delta;
    const newWeekStart = addWeeks(displayedWeekStart, delta);
    const targetDate = addDays(newWeekStart, activeDayIndex);

    const safeTargetDate =
      newWeekOffset === 0 && isAfter(targetDate, today) ? today : targetDate;

    setWeekOffset(newWeekOffset);
    setSelectedDate(format(safeTargetDate, 'yyyy-MM-dd'));
  };

  const enabledActivities = user?.enabledActivities || ['pushups', 'sports', 'water', 'protein'];
  const totalTasks = enabledActivities.length + 1; // +1 for weight
  const requiredTasks = Math.ceil(totalTasks * 0.6);

  const weekDays = Array.from({ length: 7 }, (_, i) => {
    const date = addDays(displayedWeekStart, i);
    const dateStr = format(date, 'yyyy-MM-dd');
    const dayTracking = combinedTracking[dateStr] || {};
    const isToday = isSameDay(date, today);
    const isSelected = dateStr === activeDate;

    // Check what's completed based on enabled activities
    const completedList = [];

    if (enabledActivities.includes('pushups') && (dayTracking?.pushups?.total || 0) > 0) {
      completedList.push('pushups');
    }
    if (enabledActivities.includes('sports') && countActiveSports(dayTracking.sports) > 0) {
      completedList.push('sports');
    }
    if (enabledActivities.includes('water') && (dayTracking.water || 0) >= 2000) {
      completedList.push('water');
    }
    if (enabledActivities.includes('protein') && (dayTracking?.protein || 0) >= 100) {
      completedList.push('protein');
    }
    // Weight is always tracked
    if (dayTracking.weight?.value) {
      completedList.push('weight');
    }

    const tasksCompleted = completedList.length;
    const isDone = tasksCompleted >= requiredTasks;

    return {
      label: format(date, 'EEE', { locale }),
      date: dateStr,
      isToday,
      done: isDone,
      tasksCompleted,
      isSelected,
    };
  });

  return (
    <div className="rounded-2xl bg-white/5 dark:bg-white/5 backdrop-blur-md border border-white/10 shadow-[0_6px_24px_rgba(0,0,0,0.25)] p-4 lg:p-5 transition-all duration-200">
      {/* Header */}
      <div className="flex items-start justify-between mb-3 gap-4">
        <div className="flex-1 min-w-0">
          <h2 className="text-lg lg:text-xl font-semibold text-white mb-0.5">
            {headingLabel}
          </h2>
          <p className="text-xs text-white/70">
            {t('dashboard.tapToEdit')}
          </p>
          <p className="text-xs text-white/50 mt-1">
            {t('dashboard.weekRangeLabel', { range: weekRangeLabel })}
          </p>
        </div>
        <div className="flex items-center gap-2 pt-1">
          <button
            type="button"
            onClick={() => handleWeekChange('previous')}
            className="p-2 rounded-full border border-white/10 text-white/70 hover:text-white hover:border-white/30 transition-colors"
            aria-label={t('dashboard.previousWeek')}
          >
            <span aria-hidden="true">‹</span>
          </button>
          <button
            type="button"
            onClick={() => handleWeekChange('next')}
            disabled={disableNextWeek}
            className="p-2 rounded-full border border-white/10 text-white/70 hover:text-white hover:border-white/30 transition-colors disabled:opacity-40 disabled:cursor-not-allowed"
            aria-label={t('dashboard.nextWeek')}
          >
            <span aria-hidden="true">›</span>
          </button>
        </div>
      </div>

      {/* Week Days Chips - Grid Layout (7 equal columns) */}
      <div className="grid grid-cols-7 gap-1 lg:gap-1.5">
        {weekDays.map((day) => {
          let chipClasses = 'flex items-center justify-center h-9 rounded-full border text-xs sm:text-sm font-medium transition-all cursor-pointer select-none';

          if (day.done) {
            // Completed: glow effect
            chipClasses += ' bg-emerald-500/20 border-emerald-400/70 text-emerald-100 shadow-[inset_0_0_14px_rgba(16,185,129,0.45)]';
          } else if (day.tasksCompleted > 0) {
            // Partial: amber ring
            chipClasses += ' bg-white/6 border-amber-400/60 text-amber-200';
          } else {
            // Empty
            chipClasses += ' bg-white/6 border-white/10 text-white/60';
          }

          if (day.isToday) {
            // Today: accent ring
            chipClasses += ' ring-2 ring-sky-400/70 ring-offset-2 ring-offset-transparent';
          }

          if (day.isSelected) {
            chipClasses += ' scale-105';
          }

      return (
        <button
          key={day.date}
          type="button"
          onClick={() => setSelectedDate(day.date)}
          className={chipClasses}
              title={`${day.label} - ${day.tasksCompleted}/${totalTasks} ${t('dashboard.tasks')}`}
            >
              {day.label}
            </button>
          );
        })}
      </div>

      {/* Week Progress Circles */}
      <div className="mt-3 grid grid-cols-7 gap-1">
        {weekDays.map((day) => {
          const progress = (day.tasksCompleted / totalTasks) * 100; // Dynamic total tasks

          return (
            <div key={day.date} className="flex flex-col items-center gap-0.5">
              {/* Progress Circle */}
              <div className="w-8 h-8 relative">
                <svg className="w-full h-full -rotate-90" viewBox="0 0 36 36">
                  {/* Background circle */}
                  <circle
                    cx="18"
                    cy="18"
                    r="16"
                    fill="none"
                    className="stroke-white/20"
                    strokeWidth="3"
                  />
                  {/* Progress circle */}
                  {progress > 0 && (
                    <circle
                      cx="18"
                      cy="18"
                      r="16"
                      fill="none"
                      className={`${
                        day.done
                          ? 'stroke-emerald-400'
                          : day.tasksCompleted > 0
                          ? 'stroke-amber-400'
                          : 'stroke-white/40'
                      } ${day.isToday ? 'drop-shadow-[0_0_4px_rgba(96,165,250,0.8)]' : ''}`}
                      strokeWidth="3"
                      strokeDasharray={`${progress} 100`}
                      strokeLinecap="round"
                    />
                  )}
                </svg>
                {/* Task count */}
                <div
                  className={`absolute inset-0 flex items-center justify-center text-[10px] font-semibold ${
                    day.isToday
                      ? 'text-sky-400'
                      : 'text-white'
                  }`}
                >
                  {day.tasksCompleted}
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
}
