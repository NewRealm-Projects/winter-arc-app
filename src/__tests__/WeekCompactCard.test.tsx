import { act, render } from '@testing-library/react';
import { beforeEach, afterEach, describe, expect, it, vi } from 'vitest';
import WeekCompactCard from '../components/dashboard/WeekCompactCard';
import { useStore } from '../store/useStore';
import type { SportTracking } from '../types';

const originalSelectedDate = useStore.getState().selectedDate;

describe('WeekCompactCard', () => {
  beforeEach(() => {
    vi.useFakeTimers();
    vi.setSystemTime(new Date('2024-01-10T08:00:00Z'));

    useStore.setState({
      user: {
        id: 'user-1',
        language: 'en',
        nickname: 'Tester',
        gender: 'male',
        height: 180,
        weight: 80,
        hydrationGoalLiters: 2.5,
        proteinGoalGrams: 150,
        maxPushups: 20,
        groupCode: 'grp',
        createdAt: new Date('2024-01-01T00:00:00Z'),
        pushupState: { baseReps: 10, sets: 3, restTime: 60 },
        enabledActivities: ['water', 'protein', 'sports', 'pushups'],
      },
      selectedDate: '2024-01-10',
      tracking: {
        '2024-01-08': {
          date: '2024-01-08',
          sports: {} as SportTracking,
          water: 2600,
          protein: 150,
          completed: false,
        },
        '2024-01-09': {
          date: '2024-01-09',
          sports: {} as SportTracking,
          water: 1800,
          protein: 120,
          completed: false,
        },
        '2024-01-10': {
          date: '2024-01-10',
          sports: {} as SportTracking,
          water: 2500,
          protein: 140,
          pushups: { total: 30 },
          completed: false,
        },
      },
      smartContributions: {},
    });
  });

  afterEach(() => {
    vi.useRealTimers();
    useStore.setState({
      user: null,
      tracking: {},
      smartContributions: {},
      selectedDate: originalSelectedDate,
    });
  });

  it('renders the week overview without NaN values', async () => {
    let rendered: ReturnType<typeof render> | undefined;

    await act(async () => {
      rendered = render(<WeekCompactCard />);
    });

    await act(async () => {});

    const { container } = rendered!;
    expect(container).toMatchSnapshot();
    expect(container.textContent).not.toContain('NaN');
  });
});
