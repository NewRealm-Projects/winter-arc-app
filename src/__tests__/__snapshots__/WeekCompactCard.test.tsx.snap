// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeekCompactCard > renders the week overview without NaN values 1`] = `
<div>
  <div
    class="rounded-2xl bg-white/5 dark:bg-white/5 backdrop-blur-md border border-white/10 shadow-[0_6px_24px_rgba(0,0,0,0.25)] p-4 lg:p-5 transition-all duration-200"
  >
    <div
      class="flex items-start justify-between gap-4"
    >
      <div
        class="flex-1 min-w-0"
      >
        <h2
          class="text-lg lg:text-xl font-semibold text-white mb-0.5"
        >
          This Week
        </h2>
        <p
          class="text-xs text-white/70"
        >
          Tap a day to edit or update
        </p>
        <p
          class="text-xs text-white/50 mt-1"
        >
          Jan 8 â€“ Jan 14
        </p>
      </div>
      <div
        class="flex items-center gap-2 pt-1"
      >
        <button
          aria-label="Previous week"
          class="p-2 rounded-full border border-white/10 text-white/70 hover:text-white hover:border-white/30 transition-colors"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            â€¹
          </span>
        </button>
        <button
          aria-label="Next week"
          class="p-2 rounded-full border border-white/10 text-white/70 hover:text-white hover:border-white/30 transition-colors disabled:opacity-40 disabled:cursor-not-allowed"
          disabled=""
          type="button"
        >
          <span
            aria-hidden="true"
          >
            â€º
          </span>
        </button>
      </div>
    </div>
    <div
      class="mt-4 flex flex-nowrap items-center justify-center gap-3 overflow-x-auto scroll-smooth no-scrollbar sm:grid sm:grid-cols-7 sm:items-stretch sm:justify-start sm:overflow-visible"
    >
      <button
        aria-label="Monday Â· 80%
ðŸ’§ 2.6/2.5L Â· ðŸ¥© 150/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Monday Â· 80%
ðŸ’§ 2.6/2.5L Â· ðŸ¥© 150/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Mon
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-amber-400 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="27.64601535159018"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            80
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            8
          </span>
          <span
            class="absolute -bottom-1 -right-1 flex h-5 w-5 items-center justify-center rounded-full bg-emerald-500 text-xs shadow-lg"
          >
            ðŸ”¥
          </span>
        </div>
      </button>
      <button
        aria-label="Tuesday Â· 61%
ðŸ’§ 1.8/2.5L Â· ðŸ¥© 120/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Tuesday Â· 61%
ðŸ’§ 1.8/2.5L Â· ðŸ¥© 120/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Tue
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-amber-400 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="53.90972993560085"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            61
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            9
          </span>
          <span
            class="absolute -bottom-1 -right-1 flex h-5 w-5 items-center justify-center rounded-full bg-emerald-500 text-xs shadow-lg"
          >
            ðŸ”¥
          </span>
        </div>
      </button>
      <button
        aria-label="Wednesday Â· 97%
ðŸ’§ 2.5/2.5L Â· ðŸ¥© 140/150g Â· ðŸ’ª"
        aria-pressed="true"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-[1.04]"
        title="Wednesday Â· 97%
ðŸ’§ 2.5/2.5L Â· ðŸ¥© 140/150g Â· ðŸ’ª"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Wed
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200 ring-2 ring-sky-400/70 ring-offset-2 ring-offset-transparent shadow-[0_8px_18px_rgba(56,189,248,0.35)]"
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-amber-400 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="4.146902302738539"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            97
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            10
          </span>
          <span
            class="absolute -bottom-1 -right-1 flex h-5 w-5 items-center justify-center rounded-full bg-emerald-500 text-xs shadow-lg"
          >
            ðŸ”¥
          </span>
        </div>
      </button>
      <button
        aria-label="Thursday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Thursday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Thu
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-white/25 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="138.23007675795088"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            0
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            11
          </span>
        </div>
      </button>
      <button
        aria-label="Friday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Friday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Fri
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-white/25 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="138.23007675795088"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            0
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            12
          </span>
        </div>
      </button>
      <button
        aria-label="Saturday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Saturday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Sat
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-white/25 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="138.23007675795088"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            0
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            13
          </span>
        </div>
      </button>
      <button
        aria-label="Sunday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        aria-pressed="false"
        class="group relative flex flex-shrink-0 flex-col items-center gap-2 text-white focus:outline-none focus-visible:ring-2 focus-visible:ring-sky-400/70 focus-visible:ring-offset-2 focus-visible:ring-offset-transparent transition-transform duration-200 scale-100"
        title="Sunday Â· 0%
ðŸ’§ 0/2.5L Â· ðŸ¥© 0/150g Â· ðŸ’¤"
        type="button"
      >
        <span
          class="text-[10px] uppercase tracking-wide text-white/60 group-hover:text-white/80 transition-colors"
        >
          Sun
        </span>
        <div
          class="relative flex h-16 w-16 items-center justify-center rounded-full border border-white/10 bg-white/5 transition-shadow duration-200  "
        >
          <svg
            class="absolute inset-0 h-full w-full -rotate-90"
            role="presentation"
            viewBox="0 0 48 48"
          >
            <circle
              class="stroke-white/10"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-width="4"
            />
            <circle
              class="stroke-white/25 transition-all duration-300 ease-out"
              cx="24"
              cy="24"
              fill="none"
              r="22"
              stroke-dasharray="138.23007675795088"
              stroke-dashoffset="138.23007675795088"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
          <span
            class="relative text-sm font-semibold text-white"
          >
            0
            %
          </span>
          <span
            class="absolute bottom-1 text-[10px] font-medium text-white/50"
          >
            14
          </span>
        </div>
      </button>
    </div>
    <div
      class="mt-4 flex items-center justify-between text-[11px] text-white/60"
    >
      <span
        class="flex items-center gap-1"
      >
        <span
          aria-hidden="true"
        >
          ðŸ”¥
        </span>
        <span>
          3
           
          Day Streak
        </span>
      </span>
      <span>
        Streak counts once the day hits 50%
      </span>
    </div>
  </div>
</div>
`;
